odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1811,6 +1811,13 @@ def _validate_attrs(self, node, name_manager, node_info):
                     )
 
             elif attr.startswith('decoration-'):
+                try:
+                    get_variable_names(expr)
+                except SyntaxError:
+                    msg = _(
+                        "\nThe Decoration contains syntax error:\n'%s'", expr
+                    )
+                    self._raise_view_error(msg, node)
                 vnames = get_variable_names(expr)
                 if vnames:
                     name_manager.must_have_fields(node, vnames, f"{attr}={expr}")
```

Comment:
Why do you need to do `get_variable_names` twice ? 
Should we return the syntax error message to help the user/dev ? 

```suggestion
                try:
                    vnames = get_variable_names(expr)
                except SyntaxError as e:
                    msg = _("\nThe Decoration '%s' contains syntax error:\n%s", expr, e.msg)
                    self._raise_view_error(msg, node)
                if vnames:
                    name_manager.must_have_fields(node, vnames, f"{attr}={expr}")
```

Also, IMO, we should add a test for that.